### ISSUE TITLE
Offer slimmed down version of universal template that can be extended

### ISSUE THREAD

**Issue by kingces95** [microsoft/vscode-dev-containers#1539]
Please offer a stripped-down version of the universal default image found at `https://github.com/microsoft/vscode-dev-containers/tree/main/containers/codespaces-linux` and stripped down version of `devcontainer.json`.

As a developer trying to entice my team to develop in codespace. I start by creating a demo. The demo is first constructed by creating a useful environment with the default image. Once I have a script that modifies the universal codespace, I'd like to cache those changes using the prebuilt feature. To quickly create prebuilt images containing only the tools we need, I'd like to strip out all the tools I don't need from the universal image. To do that, I first attempted to use the stripped down ubuntu image found at `https://github.com/microsoft/vscode-dev-containers/tree/main/containers/ubuntu`. That turned out to be problematic because there are differences between those images beyond just the tooling. For example:
- The user name (vscode/codespace)
- The run arguments
While the first is easy to work around, the second resulted in differences that took a few days to track down (I opened and closed [this bug](https://github.com/microsoft/vscode-dev-containers/issues/1538)). In my case, I'm trying to use `chroot` in the codespace which requires the ability to use `mount` which works in the universal image but not the ubuntu image. After some debugging, I discovered copying the runargs from the universal image to the ubuntu image fixed the problem. It seems instead of using the Ubuntu image the recommended solution is to start with the universal image and strip it down until it only has what I need. The documentation suggests this approach. First, it notes that extension is discouraged:
> Given its size, we do not recommend extending this image.

Next, it suggests cloning the repro, copying the devcontainer source, and then stripping it down by hand. This feels backwards. Instead, I'd prefer the codespaces team stripped down the universal image of all its tooling. Of course, then the question becomes where to draw the line at tooling. Certainly, anything for a specific language. Basically, just try and get the image down to a reasonable size so developers can extend it instead of cloning and stripping it. Keep all the tidbits that affect permissions, users, groups, (e.g. all that stuff in `runArgs`) and general interaction with the operating system. That I'm not sure where to draw the line is the reason I'd like the codespace team to do the stripping!

**Comment by jkeech** [comment]
Yes! One of our goals is to refactor the universal image such that you can pick and choose subsets of functionality as devcontainer "features" to drop into your own custom image. We aim to have the large universal image just be the base ubuntu devcontainer image + a set of features.

Instead of providing a large matrix of official images with varying levels of slimmed-down granularity, you'll be able to choose the right combination of config for your needs.
